src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -95,10 +95,11 @@ protected Task<ImmutableArray<CodeActionOperation>> GetPreviewOperationsAsync(Ca
 
         public void Invoke(CancellationToken cancellationToken)
         {
-            SourceProvider.WaitIndicator.Wait(CodeAction.Title, CodeAction.Message, allowCancel: true, showProgress: true, action: waitContext =>
+            SourceProvider.UIThreadOperationExecutor.Execute(CodeAction.Title, CodeAction.Message, allowCancellation: true, showProgress: true, action: context =>
             {
-                using var combinedCancellationToken = cancellationToken.CombineWith(waitContext.CancellationToken);
-                Invoke(waitContext.ProgressTracker, combinedCancellationToken.Token);
+                using var scope = context.AddScope(allowCancellation: true, CodeAction.Message);
```

Comment:
oi'm confused here.  we have both UIThreadOperation.Execute.  we pass the message and allowCancellation:truie.  But then inside we make a scope that does the same thing.  For why?

Reply:
at the very least, doc this.

Reply:
Added as a comment:

```csharp
// If we want to report progress, we need to create a scope inside the context -- the main context itself doesn't have a way
// to report progress. We pass the same allow cancellation and message so otherwise nothing changes.
```

Reply:
(and yes this took me a moment to remember so comment is most good)